assets/javascripts/test_result.js

```diff
@@ -442,26 +444,12 @@ function setupResult(state, jobid, status_url, details_url) {
   });
 
   // don't overwrite the tab if coming from the URL (ignore '#')
-  if (location.hash.length < 2 && state === "scheduled") {
+  if (location.hash.length < 2 && (state === 'scheduled' || state === 'assigned')) {
     setResultHash("#settings", true);
   }
-  // This could be easily rewritten as $.inArray
-  if ( state == "running"   ||
-       state == "uploading" ||
-       state == "assigned"  ||
-       state == "setup" ) {
+  if (state !== 'done') {
     setupRunning(jobid, status_url, details_url);
   }
-  else if (state == "scheduled") {
-    // reload when test starts
-    window.setInterval(function() {
-      $.ajax(status_url).done(function(newStatus) {
-        if (newStatus.state != 'scheduled') {
-          setTimeout(function() {location.href = location.href.replace(/#.*/, '');}, 1000);
```

Comment:
do you remember why we had that `location.href = location.href.replace(/#.*/, '')` ?

Reply:
Yes, to force showing the details tab when reloading regardless of the tab the user has currently selected. For simplicity I've removed that which means the jobs stay at the previously selected tab.

---

This is likely mainly because so far the default tab is forced to be "Settings" in pre-running jobs by setting the hash. I think the correct solution would be hiding the details and logs tabs completely until the job state is running (similarly to how the live tab is already hidden). This means the settings would be "naturally" the first (and therefore default tab) in previous states without setting the hash explicitly to `#settings`. I'm considering this for the next step (which would be getting rid of the page reload completely). For now I'd refrain from adding additional complexity for this detail